ast/ast.go

```diff
@@ -1305,6 +1307,22 @@ type CreateTable struct {
 	RowDeletionPolicy *CreateRowDeletionPolicy // optional
 }
 
+// CreateSequence is CREATE SEQUENCE statement node.
+// CREATE SEQUENCE IF NOT EXISTS {{.Name | sql}} }}
+// OPTIONS (
+//
+//	{{.Options | sqlJoin ","}}
+//
+// )
+type CreateSequence struct {
+	// pos = Create
+	Create      token.Pos // position of "CREATE" keyword
+	Rparen      token.Pos // position of ")" of OPTIONS clause
+	Name        *Ident
+	IfNotExists bool
+	Options     []*SequenceOption
```

Comment:
I'm not sure `len(Options) > 0` is actually correct. Is this right?

```suggestion
	// pos = Create
	// end = Rparen + 1

	Create      token.Pos // position of "CREATE" keyword
	Rparen      token.Pos // position of ")" of OPTIONS clause

	Name        *Ident
	IfNotExists bool
	Options     []*SequenceOption // len(Options) > 0
```

Reply:
Yes, you're right. len(Options) should be always > 0.

Reply:
Thank you. Such comments on invariants are useful for code reading.